// eslint-plugin-sxzz@0.1.0/dist/index.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true});// package.json
var version = "0.1.0";

// src/utils.ts
var hasDocs = ["prefer-string-function"];
var blobUrl = "https://github.com/sxzz/eslint-plugin-sxzz/blob/main/src/rules/";
function RuleCreator(urlCreator) {
  return function createNamedRule({
    name,
    meta,
    ...rule
  }) {
    return createRule({
      meta: {
        ...meta,
        docs: {
          ...meta.docs,
          url: urlCreator(name)
        }
      },
      ...rule
    });
  };
}
function createRule({
  create,
  defaultOptions,
  meta
}) {
  return {
    create: (context) => {
      const optionsWithDefault = context.options.map((options, index) => {
        return {
          ...defaultOptions[index] || {},
          ...options || {}
        };
      });
      return create(context, optionsWithDefault);
    },
    defaultOptions,
    meta
  };
}
var createEslintRule = RuleCreator(
  (ruleName) => hasDocs.includes(ruleName) ? `${blobUrl}${ruleName}.md` : `${blobUrl}${ruleName}.test.ts`
);

// src/rules/prefer-string-function.ts
var RULE_NAME = "prefer-string-function";
var prefer_string_function_default = createEslintRule({
  name: RULE_NAME,
  meta: {
    type: "layout",
    docs: {
      description: "Enforce using String() instead of single-member template expressions."
    },
    fixable: "code",
    schema: [],
    messages: {
      stringFunction: "Prefer `String()` instead of template expression"
    }
  },
  defaultOptions: [],
  create(context) {
    return {
      TemplateLiteral(node) {
        if (node.expressions.length === 1 && node.quasis.length === 2 && node.quasis.every((quasi) => quasi.value.raw === "")) {
          const expression = node.expressions[0];
          return context.report({
            messageId: "stringFunction",
            node,
            fix(fixer) {
              return [
                fixer.replaceTextRange(
                  [node.range[0], expression.range[0]],
                  "String("
                ),
                fixer.replaceTextRange(
                  [expression.range[1], node.range[1]],
                  ")"
                )
              ];
            }
          });
        }
      }
    };
  }
});

// src/index.ts
var plugin = {
  meta: {
    name: "sxzz",
    version
  },
  // @keep-sorted
  rules: {
    "prefer-string-function": prefer_string_function_default
  }
};
var src_default = plugin;



exports.default = src_default; exports.plugin = plugin;
