// @sxzz/eslint-config@4.6.0/dist/index.js
import _configJs from "@eslint/js";
import * as _pluginAntfu from "eslint-plugin-antfu";
import * as _pluginSxzz from "eslint-plugin-sxzz";
import configComments from "@eslint-community/eslint-plugin-eslint-comments/configs";
import * as _pluginMarkdown from "@eslint/markdown";
import tseslint from "typescript-eslint";
import * as _pluginUnicorn from "eslint-plugin-unicorn";
import * as _pluginVue from "eslint-plugin-vue";
import * as _pluginNode from "eslint-plugin-n";
import * as _pluginPerfectionist from "eslint-plugin-perfectionist";
import * as _pluginUnocss from "@unocss/eslint-plugin";
import * as _pluginPrettier from "eslint-plugin-prettier";
import * as _configPrettier from "eslint-config-prettier";
import * as _pluginUnusedImports from "eslint-plugin-unused-imports";
import * as _pluginJsdoc from "eslint-plugin-jsdoc";
import * as _pluginIgnore from "eslint-config-flat-gitignore";
import * as pluginImport from "eslint-plugin-import-x";
import * as pluginJsonc from "eslint-plugin-jsonc";
import * as pluginYml from "eslint-plugin-yml";
import * as parserVue from "vue-eslint-parser";
import * as parserYml from "yaml-eslint-parser";
import * as parserJsonc from "jsonc-eslint-parser";
import _configCommand from "eslint-plugin-command/config";
import globals from "globals";
import process from "node:process";
import { getPackageInfoSync, isPackageExists } from "local-pkg";
import { configs } from "eslint-plugin-regexp";

//#region src/plugins.ts
/* #__NO_SIDE_EFFECTS__ */ function interopDefault(m) {
	return "default" in m ? interopDefault(m.default) : m;
}
const configJs = interopDefault(_configJs);
const pluginAntfu = interopDefault(_pluginAntfu);
const pluginSxzz = interopDefault(_pluginSxzz);
const pluginMarkdown = interopDefault(_pluginMarkdown);
const pluginUnicorn = interopDefault(_pluginUnicorn);
const pluginVue = interopDefault(_pluginVue);
const pluginNode = interopDefault(_pluginNode);
const pluginPerfectionist = interopDefault(_pluginPerfectionist);
const pluginUnocss = interopDefault(_pluginUnocss);
const pluginPrettier = interopDefault(_pluginPrettier);
const configPrettier = interopDefault(_configPrettier);
const pluginUnusedImports = interopDefault(_pluginUnusedImports);
const pluginJsdoc = interopDefault(_pluginJsdoc);
const pluginIgnore = interopDefault(_pluginIgnore);
const configCommand = interopDefault(_configCommand);

//#endregion
//#region src/configs/command.ts
const command = [{
	...configCommand(),
	name: "sxzz/command"
}];

//#endregion
//#region src/configs/comments.ts
const comments = [{
	...configComments.recommended,
	name: "sxzz/comments/recommended"
}, {
	name: "sxzz/comments",
	rules: { "@eslint-community/eslint-comments/disable-enable-pair": ["error", { allowWholeFile: true }] }
}];

//#endregion
//#region src/globs.ts
const GLOB_SRC_EXT = "?([cm])[jt]s?(x)";
const GLOB_SRC = "**/*.?([cm])[jt]s?(x)";
const GLOB_JS = "**/*.?([cm])js";
const GLOB_JSX = "**/*.?([cm])jsx";
const GLOB_TS = "**/*.?([cm])ts";
const GLOB_TSX = "**/*.tsx";
const GLOB_STYLE = "**/*.{c,le,sc}ss";
const GLOB_CSS = "**/*.css";
const GLOB_LESS = "**/*.less";
const GLOB_SCSS = "**/*.scss";
const GLOB_JSON = "**/*.json";
const GLOB_JSON5 = "**/*.json5";
const GLOB_JSONC = "**/*.jsonc";
const GLOB_MARKDOWN = "**/*.md";
const GLOB_VUE = "**/*.vue";
const GLOB_YAML = "**/*.y?(a)ml";
const GLOB_HTML = "**/*.htm?(l)";
const GLOB_ALL_SRC = [
	GLOB_SRC,
	GLOB_STYLE,
	GLOB_JSON,
	GLOB_JSON5,
	GLOB_MARKDOWN,
	GLOB_VUE,
	GLOB_YAML,
	GLOB_HTML
];
const GLOB_NODE_MODULES = "**/node_modules";
const GLOB_DIST = "**/dist";
const GLOB_LOCKFILE = [
	"**/package-lock.json",
	"**/yarn.lock",
	"**/pnpm-lock.yaml",
	"**/bun.lockb"
];
const GLOB_EXCLUDE = [
	GLOB_NODE_MODULES,
	GLOB_DIST,
	...GLOB_LOCKFILE,
	"**/output",
	"**/coverage",
	"**/temp",
	"**/fixtures",
	"**/.vitepress/cache",
	"**/.nuxt",
	"**/.vercel",
	"**/.changeset",
	"**/.idea",
	"**/.output",
	"**/.vite-inspect",
	"**/.nitro",
	"**/CHANGELOG*.md",
	"**/*.min.*",
	"**/LICENSE*",
	"**/__snapshots__",
	"**/auto-import?(s).d.ts",
	"**/components.d.ts"
];

//#endregion
//#region src/configs/ignores.ts
const ignores = [{
	ignores: GLOB_EXCLUDE,
	name: "sxzz/global-ignores"
}, {
	...pluginIgnore({ strict: false }),
	name: "sxzz/gitignore"
}];

//#endregion
//#region src/configs/imports.ts
const imports = [{
	name: "sxzz/imports",
	plugins: {
		antfu: pluginAntfu,
		import: pluginImport
	},
	rules: {
		"antfu/import-dedupe": "error",
		"import/first": "error",
		"import/no-default-export": "error",
		"import/no-duplicates": "error",
		"import/no-mutable-exports": "error",
		"import/no-named-default": "error",
		"import/no-self-import": "error",
		"import/no-webpack-loader-syntax": "error"
	}
}];

//#endregion
//#region src/env.ts
const isInEditor = !!((process.env.VSCODE_PID || process.env.VSCODE_CWD || process.env.JETBRAINS_IDE || process.env.VIM) && !process.env.CI);
const hasTypeScript = isPackageExists("typescript");
const hasVue = isPackageExists("vue") || isPackageExists("nuxt") || isPackageExists("vitepress") || isPackageExists("@slidev/cli");
const hasUnocss = isPackageExists("unocss") || isPackageExists("@unocss/webpack") || isPackageExists("@unocss/nuxt");

//#endregion
//#region src/configs/javascript.ts
const restrictedSyntaxJs = [
	"ForInStatement",
	"LabeledStatement",
	"WithStatement"
];
const javascript = [{
	...configJs.configs.recommended,
	name: "sxzz/js/recommended"
}, {
	languageOptions: {
		globals: {
			...globals.browser,
			...globals.es2021,
			...globals.node
		},
		parserOptions: {
			ecmaFeatures: { jsx: true },
			sourceType: "module"
		},
		sourceType: "module"
	},
	name: "sxzz/js",
	plugins: {
		sxzz: pluginSxzz,
		"unused-imports": pluginUnusedImports
	},
	rules: {
		"array-callback-return": "error",
		"block-scoped-var": "error",
		"dot-notation": "warn",
		eqeqeq: ["error", "smart"],
		"no-alert": "warn",
		"no-console": ["warn", { allow: [
			"warn",
			"error",
			"info",
			"clear"
		] }],
		"no-debugger": "warn",
		"no-duplicate-imports": "error",
		"no-empty": ["error", { allowEmptyCatch: true }],
		"no-fallthrough": ["warn", { commentPattern: String.raw`break[\s\w]*omitted` }],
		"no-inner-declarations": "error",
		"no-lonely-if": "error",
		"no-multi-str": "error",
		"no-restricted-syntax": ["error", ...restrictedSyntaxJs],
		"no-unused-expressions": ["error", {
			allowShortCircuit: true,
			allowTaggedTemplates: true,
			allowTernary: true
		}],
		"no-unused-vars": "off",
		"no-void": "error",
		"object-shorthand": [
			"error",
			"always",
			{
				avoidQuotes: true,
				ignoreConstructors: false
			}
		],
		"prefer-arrow-callback": ["error", {
			allowNamedFunctions: false,
			allowUnboundThis: true
		}],
		"prefer-const": ["warn", {
			destructuring: "all",
			ignoreReadBeforeAssign: true
		}],
		"prefer-exponentiation-operator": "error",
		"prefer-regex-literals": ["error", { disallowRedundantWrapping: true }],
		"prefer-rest-params": "error",
		"prefer-spread": "error",
		"prefer-template": "error",
		"require-await": "error",
		"sxzz/prefer-string-function": "warn",
		"unicode-bom": ["error", "never"],
		"unused-imports/no-unused-imports": isInEditor ? "off" : "error",
		"unused-imports/no-unused-vars": ["error", {
			args: "after-used",
			ignoreRestSiblings: true
		}],
		"use-isnan": ["error", {
			enforceForIndexOf: true,
			enforceForSwitchCase: true
		}],
		"valid-typeof": ["error", { requireStringLiterals: true }],
		"vars-on-top": "error"
	}
}];

//#endregion
//#region src/configs/jsdoc.ts
const jsdoc = [{
	name: "sxzz/jsdoc",
	plugins: { jsdoc: pluginJsdoc },
	rules: {
		"jsdoc/check-access": "warn",
		"jsdoc/check-param-names": "warn",
		"jsdoc/check-property-names": "warn",
		"jsdoc/check-types": "warn",
		"jsdoc/empty-tags": "warn",
		"jsdoc/implements-on-classes": "warn",
		"jsdoc/no-defaults": "warn",
		"jsdoc/no-multi-asterisks": "warn",
		"jsdoc/require-param-name": "warn",
		"jsdoc/require-property": "warn",
		"jsdoc/require-property-description": "warn",
		"jsdoc/require-property-name": "warn",
		"jsdoc/require-returns-check": "warn",
		"jsdoc/require-returns-description": "warn",
		"jsdoc/require-yields-check": "warn"
	}
}];

//#endregion
//#region src/configs/jsonc.ts
const jsonc = [{
	files: [
		GLOB_JSON,
		GLOB_JSON5,
		GLOB_JSONC
	],
	languageOptions: { parser: parserJsonc },
	name: "sxzz/json",
	plugins: { jsonc: pluginJsonc },
	rules: {
		...pluginJsonc.configs["recommended-with-jsonc"].rules,
		"jsonc/quote-props": "off",
		"jsonc/quotes": "off"
	}
}];

//#endregion
//#region src/configs/markdown.ts
const markdown = [...pluginMarkdown.configs.processor.map((config) => ({
	...config,
	name: `sxzz/${config.name || "anonymous"}`
})), {
	files: [`${GLOB_MARKDOWN}/${GLOB_SRC}`, `${GLOB_MARKDOWN}/${GLOB_VUE}`],
	name: "sxzz/markdown-rules",
	rules: {
		"@typescript-eslint/comma-dangle": "off",
		"@typescript-eslint/consistent-type-imports": "off",
		"@typescript-eslint/no-extraneous-class": "off",
		"@typescript-eslint/no-namespace": "off",
		"@typescript-eslint/no-redeclare": "off",
		"@typescript-eslint/no-require-imports": "off",
		"@typescript-eslint/no-unused-expressions": "off",
		"@typescript-eslint/no-unused-vars": "off",
		"@typescript-eslint/no-use-before-define": "off",
		"no-alert": "off",
		"no-console": "off",
		"no-restricted-imports": "off",
		"no-undef": "off",
		"no-unused-expressions": "off",
		"no-unused-vars": "off",
		"node/prefer-global/buffer": "off",
		"node/prefer-global/process": "off",
		"unused-imports/no-unused-imports": "off",
		"unused-imports/no-unused-vars": "off"
	}
}];

//#endregion
//#region src/configs/node.ts
const node = [{
	name: "sxzz/node",
	plugins: { node: pluginNode },
	rules: {
		"node/handle-callback-err": ["error", "^(err|error)$"],
		"node/no-deprecated-api": "error",
		"node/no-exports-assign": "error",
		"node/no-new-require": "error",
		"node/no-path-concat": "error",
		"node/no-unsupported-features/es-builtins": "error",
		"node/prefer-global/buffer": ["error", "never"],
		"node/prefer-global/process": ["error", "never"],
		"node/process-exit-as-throw": "error"
	}
}];

//#endregion
//#region src/configs/prettier.ts
const prettierConflictRules = { ...configPrettier.rules };
delete prettierConflictRules["vue/html-self-closing"];
const prettier = [{
	name: "sxzz/prettier",
	plugins: { prettier: pluginPrettier },
	rules: {
		...prettierConflictRules,
		...pluginPrettier.configs.recommended.rules,
		"prettier/prettier": "warn"
	}
}];

//#endregion
//#region src/configs/regexp.ts
const regexp = [{
	...configs["flat/recommended"],
	name: "sxzz/regexp"
}];

//#endregion
//#region src/configs/sort.ts
const sortPackageJson = [{
	files: ["**/package.json"],
	name: "sxzz/sort/package.json",
	rules: {
		"jsonc/sort-array-values": ["error", {
			order: { type: "asc" },
			pathPattern: "^files$"
		}],
		"jsonc/sort-keys": [
			"error",
			{
				order: [
					"name",
					"version",
					"private",
					"packageManager",
					"description",
					"type",
					"keywords",
					"license",
					"homepage",
					"bugs",
					"repository",
					"author",
					"contributors",
					"funding",
					"files",
					"main",
					"module",
					"types",
					"exports",
					"typesVersions",
					"sideEffects",
					"unpkg",
					"jsdelivr",
					"browser",
					"bin",
					"man",
					"directories",
					"publishConfig",
					"scripts",
					"peerDependencies",
					"peerDependenciesMeta",
					"optionalDependencies",
					"dependencies",
					"devDependencies",
					"engines",
					"config",
					"overrides",
					"pnpm",
					"husky",
					"lint-staged",
					"eslintConfig",
					"prettier"
				],
				pathPattern: "^$"
			},
			{
				order: { type: "asc" },
				pathPattern: "^(?:dev|peer|optional|bundled)?[Dd]ependencies(Meta)?$"
			},
			{
				order: [
					"types",
					"require",
					"import",
					"default"
				],
				pathPattern: "^exports.*$"
			},
			{
				order: { type: "asc" },
				pathPattern: "^(?:resolutions|overrides|pnpm.overrides)$"
			}
		]
	}
}];
const sortTsconfig = [{
	files: ["**/tsconfig.json", "**/tsconfig.*.json"],
	name: "sxzz/sort/tsconfig",
	rules: { "jsonc/sort-keys": [
		"error",
		{
			order: [
				"extends",
				"compilerOptions",
				"references",
				"files",
				"include",
				"exclude"
			],
			pathPattern: "^$"
		},
		{
			order: [
				"incremental",
				"composite",
				"tsBuildInfoFile",
				"disableSourceOfProjectReferenceRedirect",
				"disableSolutionSearching",
				"disableReferencedProjectLoad",
				"target",
				"jsx",
				"jsxFactory",
				"jsxFragmentFactory",
				"jsxImportSource",
				"lib",
				"moduleDetection",
				"noLib",
				"reactNamespace",
				"useDefineForClassFields",
				"emitDecoratorMetadata",
				"experimentalDecorators",
				"baseUrl",
				"rootDir",
				"rootDirs",
				"customConditions",
				"module",
				"moduleResolution",
				"moduleSuffixes",
				"noResolve",
				"paths",
				"resolveJsonModule",
				"resolvePackageJsonExports",
				"resolvePackageJsonImports",
				"typeRoots",
				"types",
				"allowArbitraryExtensions",
				"allowImportingTsExtensions",
				"allowUmdGlobalAccess",
				"allowJs",
				"checkJs",
				"maxNodeModuleJsDepth",
				"strict",
				"strictBindCallApply",
				"strictFunctionTypes",
				"strictNullChecks",
				"strictPropertyInitialization",
				"allowUnreachableCode",
				"allowUnusedLabels",
				"alwaysStrict",
				"exactOptionalPropertyTypes",
				"noFallthroughCasesInSwitch",
				"noImplicitAny",
				"noImplicitOverride",
				"noImplicitReturns",
				"noImplicitThis",
				"noPropertyAccessFromIndexSignature",
				"noUncheckedIndexedAccess",
				"noUnusedLocals",
				"noUnusedParameters",
				"useUnknownInCatchVariables",
				"declaration",
				"declarationDir",
				"declarationMap",
				"downlevelIteration",
				"emitBOM",
				"emitDeclarationOnly",
				"importHelpers",
				"importsNotUsedAsValues",
				"inlineSourceMap",
				"inlineSources",
				"isolatedDeclarations",
				"mapRoot",
				"newLine",
				"noEmit",
				"noEmitHelpers",
				"noEmitOnError",
				"outDir",
				"outFile",
				"preserveConstEnums",
				"preserveValueImports",
				"removeComments",
				"sourceMap",
				"sourceRoot",
				"stripInternal",
				"allowSyntheticDefaultImports",
				"esModuleInterop",
				"forceConsistentCasingInFileNames",
				"isolatedModules",
				"preserveSymlinks",
				"verbatimModuleSyntax",
				"skipDefaultLibCheck",
				"skipLibCheck"
			],
			pathPattern: "^compilerOptions$"
		}
	] }
}];
const sortImports = [{
	name: "sxzz/sort/imports",
	plugins: { perfectionist: pluginPerfectionist },
	rules: {
		"perfectionist/sort-imports": ["warn", {
			groups: [
				"builtin",
				"external",
				"internal",
				"internal-type",
				"parent",
				"parent-type",
				"sibling",
				"sibling-type",
				"index",
				"index-type",
				"object",
				"type",
				"side-effect",
				"side-effect-style"
			],
			internalPattern: ["^[~@#]/.*"],
			newlinesBetween: "ignore"
		}],
		"perfectionist/sort-named-exports": ["warn", { groupKind: "values-first" }],
		"perfectionist/sort-named-imports": ["warn", { groupKind: "values-first" }]
	}
}];

//#endregion
//#region src/configs/special-cases.ts
const specialCases = [
	{
		files: ["**/scripts/*", "**/cli.*"],
		name: "sxzz/special/cli",
		rules: { "no-console": "off" }
	},
	{
		files: [`**/*.{test,spec}.${GLOB_SRC_EXT}`],
		name: "sxzz/special/tests",
		rules: {
			"no-unused-expressions": "off",
			"unicorn/consistent-function-scoping": "off"
		}
	},
	{
		files: [
			`**/*config*.${GLOB_SRC_EXT}`,
			`**/{views,pages,routes,middleware,plugins,api,modules}/${GLOB_SRC}`,
			`**/{index,vite,esbuild,rollup,rolldown,webpack,rspack,farm}.${GLOB_SRC_EXT}`,
			"**/*.d.ts",
			`${GLOB_MARKDOWN}/**`,
			"**/.prettierrc*"
		],
		name: "sxzz/special/allow-default-export",
		plugins: { import: pluginImport },
		rules: { "import/no-default-export": "off" }
	}
];

//#endregion
//#region src/configs/typescript.ts
const typescriptCore = tseslint.config({
	extends: [...tseslint.configs.recommended],
	files: [GLOB_TS, GLOB_TSX],
	name: "sxzz/typescript",
	rules: {
		"@typescript-eslint/ban-ts-comment": "off",
		"@typescript-eslint/consistent-type-assertions": ["error", {
			assertionStyle: "as",
			objectLiteralTypeAssertions: "allow-as-parameter"
		}],
		"@typescript-eslint/consistent-type-imports": ["error", {
			disallowTypeAnnotations: false,
			fixStyle: "inline-type-imports"
		}],
		"@typescript-eslint/method-signature-style": ["error", "property"],
		"@typescript-eslint/no-empty-object-type": "off",
		"@typescript-eslint/no-explicit-any": "off",
		"@typescript-eslint/no-import-type-side-effects": "error",
		"@typescript-eslint/no-non-null-assertion": "off",
		"@typescript-eslint/no-redeclare": "error",
		"@typescript-eslint/no-unsafe-function-type": "off",
		"@typescript-eslint/no-unused-expressions": ["error", {
			allowShortCircuit: true,
			allowTaggedTemplates: true,
			allowTernary: true
		}],
		"@typescript-eslint/no-unused-vars": "off",
		"@typescript-eslint/prefer-as-const": "warn",
		"@typescript-eslint/prefer-literal-enum-member": ["error", { allowBitwiseExpressions: true }],
		"no-restricted-syntax": [
			"error",
			...restrictedSyntaxJs,
			"TSEnumDeclaration[const=true]"
		]
	}
});
const typescript = [
	...typescriptCore,
	{
		files: ["**/*.d.ts"],
		name: "sxzz/typescript/dts-rules",
		rules: {
			"eslint-comments/no-unlimited-disable": "off",
			"import/no-duplicates": "off",
			"no-restricted-syntax": "off",
			"unused-imports/no-unused-vars": "off"
		}
	},
	{
		files: [GLOB_JS, "**/*.cjs"],
		name: "sxzz/typescript/cjs-rules",
		rules: { "@typescript-eslint/no-require-imports": "off" }
	}
];

//#endregion
//#region src/configs/unicorn.ts
const unicorn = [{
	name: "sxzz/unicorn",
	plugins: { unicorn: pluginUnicorn },
	rules: {
		"unicorn/catch-error-name": "error",
		"unicorn/consistent-empty-array-spread": "error",
		"unicorn/consistent-existence-index-check": "error",
		"unicorn/consistent-function-scoping": ["error", { checkArrowFunctions: false }],
		"unicorn/custom-error-definition": "error",
		"unicorn/error-message": "error",
		"unicorn/escape-case": "error",
		"unicorn/filename-case": ["error", {
			cases: {
				kebabCase: true,
				pascalCase: true
			},
			ignore: [/^[A-Z]+\..*$/, /import_map\.json/]
		}],
		"unicorn/new-for-builtins": "error",
		"unicorn/no-array-method-this-argument": "error",
		"unicorn/no-array-push-push": "error",
		"unicorn/no-await-in-promise-methods": "error",
		"unicorn/no-console-spaces": "error",
		"unicorn/no-for-loop": "error",
		"unicorn/no-hex-escape": "error",
		"unicorn/no-instanceof-array": "error",
		"unicorn/no-invalid-remove-event-listener": "error",
		"unicorn/no-lonely-if": "error",
		"unicorn/no-negation-in-equality-check": "error",
		"unicorn/no-new-array": "error",
		"unicorn/no-new-buffer": "error",
		"unicorn/no-single-promise-in-promise-methods": "error",
		"unicorn/no-static-only-class": "error",
		"unicorn/no-unnecessary-await": "error",
		"unicorn/no-zero-fractions": `error`,
		"unicorn/prefer-add-event-listener": "error",
		"unicorn/prefer-array-find": "error",
		"unicorn/prefer-array-flat-map": "error",
		"unicorn/prefer-array-index-of": "error",
		"unicorn/prefer-array-some": "error",
		"unicorn/prefer-at": "error",
		"unicorn/prefer-blob-reading-methods": "error",
		"unicorn/prefer-date-now": "error",
		"unicorn/prefer-dom-node-append": "error",
		"unicorn/prefer-dom-node-dataset": "error",
		"unicorn/prefer-dom-node-remove": "error",
		"unicorn/prefer-dom-node-text-content": "error",
		"unicorn/prefer-includes": "error",
		"unicorn/prefer-keyboard-event-key": "error",
		"unicorn/prefer-math-min-max": "error",
		"unicorn/prefer-math-trunc": "error",
		"unicorn/prefer-modern-dom-apis": "error",
		"unicorn/prefer-modern-math-apis": "error",
		"unicorn/prefer-negative-index": "error",
		"unicorn/prefer-node-protocol": "error",
		"unicorn/prefer-number-properties": "error",
		"unicorn/prefer-optional-catch-binding": "error",
		"unicorn/prefer-prototype-methods": "error",
		"unicorn/prefer-query-selector": "error",
		"unicorn/prefer-reflect-apply": "error",
		"unicorn/prefer-regexp-test": "error",
		"unicorn/prefer-string-raw": "error",
		"unicorn/prefer-string-replace-all": "error",
		"unicorn/prefer-string-slice": "error",
		"unicorn/prefer-string-starts-ends-with": "error",
		"unicorn/prefer-string-trim-start-end": "error",
		"unicorn/prefer-type-error": "error",
		"unicorn/throw-new-error": "error"
	}
}];

//#endregion
//#region src/configs/unocss.ts
const unocss = [{
	...pluginUnocss.configs.flat,
	name: "sxzz/unocss"
}];

//#endregion
//#region src/configs/vue.ts
function getVueVersion() {
	const pkg = getPackageInfoSync("vue", { paths: [process.cwd()] });
	if (pkg && typeof pkg.version === "string" && !Number.isNaN(+pkg.version[0])) return +pkg.version[0];
	return 3;
}
const isVue3 = getVueVersion() === 3;
const reactivityTransform = [{
	languageOptions: { globals: {
		$: "readonly",
		$$: "readonly",
		$computed: "readonly",
		$customRef: "readonly",
		$ref: "readonly",
		$shallowRef: "readonly",
		$toRef: "readonly"
	} },
	name: "sxzz/vue/reactivity-transform",
	plugins: { vue: pluginVue },
	rules: { "vue/no-setup-props-reactivity-loss": "off" }
}];
const vueCustomRules = {
	"vue/block-order": ["error", { order: [
		"script",
		"template",
		"style"
	] }],
	"vue/custom-event-name-casing": ["error", "camelCase"],
	"vue/eqeqeq": ["error", "smart"],
	"vue/html-self-closing": ["error", {
		html: {
			component: "always",
			normal: "always",
			void: "any"
		},
		math: "always",
		svg: "always"
	}],
	"vue/max-attributes-per-line": "off",
	"vue/multi-word-component-names": "off",
	"vue/no-constant-condition": "warn",
	"vue/no-empty-pattern": "error",
	"vue/no-loss-of-precision": "error",
	"vue/no-unused-refs": "error",
	"vue/no-useless-v-bind": "error",
	"vue/no-v-html": "off",
	"vue/object-shorthand": [
		"error",
		"always",
		{
			avoidQuotes: true,
			ignoreConstructors: false
		}
	],
	"vue/one-component-per-file": "off",
	"vue/padding-line-between-blocks": ["error", "always"],
	"vue/prefer-template": "error",
	"vue/require-default-prop": "off",
	"vue/require-prop-types": "off"
};
const vue3Rules = {
	...pluginVue.configs.base.rules,
	...pluginVue.configs["vue3-essential"].rules,
	...pluginVue.configs["vue3-strongly-recommended"].rules,
	...pluginVue.configs["vue3-recommended"].rules
};
const vue2Rules = {
	...pluginVue.configs.base.rules,
	...pluginVue.configs.essential.rules,
	...pluginVue.configs["strongly-recommended"].rules,
	...pluginVue.configs.recommended.rules
};
delete vue2Rules["vue/component-tags-order"];
delete vue3Rules["vue/component-tags-order"];
const vueTs = typescriptCore.filter((config) => config.name !== "typescript-eslint/base").map((config) => {
	return {
		...config,
		files: [GLOB_VUE],
		name: `sxzz/vue/${config.name?.replace("sxzz/", "") || "anonymous"}`
	};
});
const vue = [
	...vueTs,
	{
		files: [GLOB_VUE],
		languageOptions: {
			parser: parserVue,
			parserOptions: {
				ecmaFeatures: { jsx: true },
				extraFileExtensions: [".vue"],
				parser: "@typescript-eslint/parser",
				sourceType: "module"
			}
		},
		name: "sxzz/vue",
		plugins: {
			"@typescript-eslint": tseslint.plugin,
			vue: pluginVue
		},
		processor: pluginVue.processors[".vue"],
		rules: {
			...isVue3 ? vue3Rules : vue2Rules,
			...vueCustomRules
		}
	},
	...reactivityTransform
];

//#endregion
//#region src/configs/yml.ts
const yml = [{
	files: [GLOB_YAML],
	languageOptions: { parser: parserYml },
	name: "sxzz/yaml",
	plugins: { yml: pluginYml },
	rules: {
		...pluginYml.configs.standard.rules,
		...pluginYml.configs.prettier.rules,
		"yml/no-empty-mapping-value": "off"
	}
}];

//#endregion
//#region src/presets.ts
const presetJavaScript = [
	...ignores,
	...javascript,
	...comments,
	...imports,
	...unicorn,
	...node,
	...jsdoc,
	...regexp
];
const presetJsonc = [
	...jsonc,
	...sortPackageJson,
	...sortTsconfig
];
const presetLangsExtensions = [
	...markdown,
	...yml,
	...presetJsonc
];
const presetBasic = [
	...presetJavaScript,
	...typescript,
	...sortImports
];
const presetAll = [
	...presetBasic,
	...presetLangsExtensions,
	...vue,
	...unocss,
	...prettier
];
function sxzz(config = [], { command: enableCommand = true, markdown: enableMarkdown = true, prettier: enablePrettier = true, unocss: enableUnocss = hasUnocss, vue: enableVue = hasVue } = {}) {
	const configs$1 = [
		...presetBasic,
		...yml,
		...presetJsonc
	];
	if (enableVue) configs$1.push(...vue);
	if (enableMarkdown) configs$1.push(...markdown);
	if (enableUnocss) configs$1.push(...unocss);
	if (enablePrettier) configs$1.push(...prettier);
	if (enableCommand) configs$1.push(...command);
	if (Object.keys(config).length > 0) configs$1.push(...Array.isArray(config) ? config : [config]);
	configs$1.push(...specialCases);
	return configs$1;
}

//#endregion
export { GLOB_ALL_SRC, GLOB_CSS, GLOB_DIST, GLOB_EXCLUDE, GLOB_HTML, GLOB_JS, GLOB_JSON, GLOB_JSON5, GLOB_JSONC, GLOB_JSX, GLOB_LESS, GLOB_LOCKFILE, GLOB_MARKDOWN, GLOB_NODE_MODULES, GLOB_SCSS, GLOB_SRC, GLOB_SRC_EXT, GLOB_STYLE, GLOB_TS, GLOB_TSX, GLOB_VUE, GLOB_YAML, presetAll as all, presetBasic as basic, command, comments, configCommand, configComments, configJs, configPrettier, getVueVersion, hasTypeScript, hasUnocss, hasVue, ignores, imports, isInEditor, javascript, jsdoc, jsonc, markdown, node, parserJsonc, parserVue, parserYml, pluginAntfu, pluginIgnore, pluginImport, pluginJsdoc, pluginJsonc, pluginMarkdown, pluginNode, pluginPerfectionist, pluginPrettier, pluginSxzz, pluginUnicorn, pluginUnocss, pluginUnusedImports, pluginVue, pluginYml, presetAll, presetBasic, presetJavaScript, presetJsonc, presetLangsExtensions, prettier, reactivityTransform, regexp, restrictedSyntaxJs, sortImports, sortPackageJson, sortTsconfig, specialCases, sxzz, tseslint, typescript, typescriptCore, unicorn, unocss, vue, yml };
